### DISCUSSION TITLE
Key Gen general

### DISCUSSION THREAD

**Discussion by charely6** [charely6/keygen#2] [General]
General discussions for now 
current topic includes Medco 3 and general bug fixes.

**Comment by charely6** [comment]
So I see what's wrong with the generated schlage.gen.scad file.
If you open it and uncomment the polygon function at the bottom for one of the outlines (either 5pin or 6pin) and then do F5 and compare that to doing the same thing Medco.gen.scad or one of the others. You will notice that the outline is rotated by 90 degrees to the others. I suspect that this is due to some update for inkscape because technically the paths2openscad.py is an inkscape extension. by uncommenting the other polygons in that file those all "line up" with their paths from the svg file. I'm looking at the svg now to try and fix this properly.

**Reply by adventuredan** [reply]
Checked and confirmed Schlage is working fine now. I just noticed that the kwikset and best scads have this issue as well. i'll check to see if those can be fixed similarly. 

also just curious, what version of openscad are you using? i've found that with versions of openscad after 2015.03, many of the keyways (at least in the medeco SVG) will preview render just fine, but on full render throw the following error:

ERROR: The given mesh is not closed! Unable to convert to CGAL_Nef_Polyhedron.

I've tested on all version after 2015 and they all have this issue on certain wardings.

**Reply by adventuredan** [reply]
so almost a year later i have solved this issue. so in openscad versions past 2015.3 the handling of vertices for mesh closure seems to have gotten tighter and a lot of the svg paths for certain wardings and outlines have unneccesary vertices and possibly closing vertices that were not actually dead on eachother. by running simplify paths on the 1644 warding which i could not get to render, the extra vertices were removed with no detectable impact to geometry accuracy. and most importantly the key now renders properly. I will now go through and test all wardings outlines and engravings to ensure that no others are having this issue and clean up the related inkscape paths then push to my fork and pull request.

**Comment by charely6** [comment]
I think I have fixed the schlage.svg issue here in my repo, and I have added some new things to keygen.scad in the newFeatures branch enjoy.

**Reply by adventuredan** [reply]
Just curious, what happened to the Schlage Primus SVGs ? they appear to be not visible anymore however i see the paths are still somewhere in the file.

**Comment by adventuredan** [comment]
just did my first pull request to add the M3 module and update some various spec specifics in the other medeco modules. Should i be doing the pull request on the new features branch instead? still learning how githubs structure works.

**Reply by charely6** [reply]
well I would say do it to the master branch, my thoughts on the new feature branch was changes to the core functions. I'll check my open scad version later when I get home and see what I can

I'll also go thru those other svgs and see what I can do

**Reply by charely6** [reply]
I probably won't actually get a chance to do it until at least tomorrow maybe Monday

**Comment by adventuredan** [comment]
just fixed all the medeco issues in classic, biaxial, and m3. gonna start seeing what i can do about kwikset and best as these seem to be broken as well
